platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/MapPropertyIntegrationTest.groovy

```diff
@@ -788,4 +788,39 @@ task thing {
         then:
         failureCauseContains("Circular evaluation detected")
     }
+
+    def "entry provider carries task dependencies"() {
+        buildFile """
+            def barTask = tasks.register("bar")
+            def barString = barTask.map { "bar" }
+
+            def bazTask = tasks.register("baz")
+            def bazString = bazTask.map { "baz" }
+
+            def map = objects.mapProperty(String, String)
+            map.put("42", barString)
+            map.put("1", bazString)
+
+            abstract class FooTask extends DefaultTask {
+                @Input
+                abstract Property<String> getEntry()
+
+                @TaskAction
+                void run() {
+                    println("Entry is \${entry.get()}")
+                }
+            }
+
+            tasks.register("foo", FooTask) {
+                getEntry().set(map.getting("42"))
+            }
+        """
+
+        when:
+        run "foo"
+
+        then: "all MapProperty's dependencies are evaluated"
+        outputContains("Entry is bar")
+        result.assertTasksExecuted(":bar", ":baz", ":foo")
```

Comment:
💭 Should we add a comment here, saying that it's not the ideal behavior, and we eventually want to track dependencies in a more fine-grained way?

In a more general sense, I sometimes miss such comments in the code written years ago, when it's unclear if there was a reason why the implementation ended up this way. Not sure if tests are the best place, but I think it's good to note it at least somewhere for the next person taking a look at it.

Reply:
I believe it going to be useful. I'll add it